### PR TITLE
Fix paragraph deletion problems

### PR DESCRIPTION
**Pull request by jotare**
### Description
Paragraph deletion is currently done by paragraph_id, this sometimes lead to not deleted paragraphs. This PR adds the capability to delete paragraphs by field_id

### How was this PR tested?
Integration tests

### CODE REVIEW

File: nucliadb/src/nucliadb/ingest/orm/brain.py [hunk 1]
```diff
@@ -229,12 +229,20 @@ def apply_field_metadata(
 
     def delete_metadata(self, field_key: str, metadata: FieldComputedMetadata):
         for subfield, metadata_split in metadata.split_metadata.items():
+            self.brain.paragraphs_to_delete.append(
+                ids.FieldId(rid=self.rid, field_id=field_key, subfield_id=subfield).full()
+            )
+            # TODO: Bw/c, remove this when the new paragraph deletion by field
```
Comment (lferran) [inline nucliadb/src/nucliadb/ingest/orm/brain.py:235]: what is paragraph deletion by field?
Isn't that what we have implemented in this pr?

File: nucliadb/tests/nucliadb/integration/test_deletion.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        fcmw = FieldComputedMetadataWrapper()
+        fcmw.field.CopyFrom(pbfield)
+        fcmw.metadata.metadata.paragraphs.append(Paragraph(start=0, end=len(field_data.extracted_text)))
+        bm.field_metadata.append(fcmw)
+
+    await inject_message(nucliadb_grpc, bm)
+
+    await asyncio.sleep(0.5)  # wait for a while until reader gets updated
+
+    # Check that searching for original texts does not return any results
```
Comment (lferran) [inline nucliadb/tests/nucliadb/integration/test_deletion.py:230]: wrong comment

File: nucliadb/tests/nucliadb/integration/test_deletion.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    assert resp.status_code == 200
+    resp_json = resp.json()
+    assert len(resp_json["resources"]) == 1
+    _rid, resource = resp_json["resources"].popitem()
+    assert rid == _rid
+    fields = resource["fields"]
+    assert len(fields) == 2
+    assert list(sorted(fields.keys())) == ["/a/summary", "/a/title"]
+
+    # Check that searching for extracted texts returns all fields
```
Comment (lferran) [inline nucliadb/tests/nucliadb/integration/test_deletion.py:249]: comment not precise enough

**Review by lferran — APPROVED** [review]

File: nucliadb/src/nucliadb/ingest/orm/brain.py (hunk 1 above)
Comment (jotare) [reply nucliadb/src/nucliadb/ingest/orm/brain.py:235]: Yeah, maybe the new is not the best word. But yes, is the one implemented in this PR

File: nucliadb/src/nucliadb/ingest/orm/brain.py [hunk 4]
```diff
@@ -229,12 +229,20 @@ def apply_field_metadata(
...
                 self.brain.paragraphs_to_delete.append(
                     f"{self.rid}/{field_key}/{subfield}/{paragraph.start}-{paragraph.end}"
                 )
 
         for paragraph in metadata.metadata.paragraphs:
+            self.brain.paragraphs_to_delete.append(ids.FieldId(rid=self.rid, field_id=field_key).full())
+            # TODO: Bw/c, remove this when the new paragraph deletion by field
+            # is promoted
             self.brain.paragraphs_to_delete.append(
                 f"{self.rid}/{field_key}/{paragraph.start}-{paragraph.end}"
```
[the remark this answers is not in the document]
Comment (jotare) [reply nucliadb/src/nucliadb/ingest/orm/brain.py:247]: I've changed the index to expect a mix of paragraph_id and field_id. It tries to remove from both terms. I think it's the safer way to promote this without breaking anything. Although, if we plan to do a rollover, maybe it's not really important

File: nucliadb/tests/nucliadb/integration/test_deletion.py (hunk 2 above)
Comment (jotare) [reply nucliadb/tests/nucliadb/integration/test_deletion.py:230]: Fixing

File: nucliadb/tests/nucliadb/integration/test_deletion.py (hunk 3 above)
Comment (jotare) [reply nucliadb/tests/nucliadb/integration/test_deletion.py:249]: Fixing

**Pull request merged**